File: upl-purchase-sdk-php/src/Model/PaymentMethod.php
<?php

/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */

declare(strict_types=1);

namespace Unzer\PayLater\Model;

use InvalidArgumentException;

use function array_key_exists;
use function sprintf;

class PaymentMethod implements Enum
{
    public const DIRECT_DEBIT = 'DIRECT_DEBIT';
    public const BANK_TRANSFER = 'BANK_TRANSFER';

    public const ALLOWED_TYPES = [
        'DIRECT_DEBIT' => self::DIRECT_DEBIT,
        'BANK_TRANSFER' => self::BANK_TRANSFER,
    ];

    /**
     * @var string
     */
    private $value;

    /**
      * @param string $value
      * @throws InvalidArgumentException
      */
    public function __construct(string $value)
    {
        if (!array_key_exists($value, self::ALLOWED_TYPES)) {
            throw new InvalidArgumentException(sprintf('Unexpected value "%s"', $value));
        }
        $this->value = $value;
    }

    /**
     * @return string
     */
    public function getValue(): string
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->value;
    }
}

File: upl-purchase-sdk-java/src/main/java/com/unzer/paylater/model/PaymentMethod.java
/**
 * Purchase API
 * @copyright Copyright (c) 2020 Unzer Pay Later
 * @license see LICENSE.TXT
 *
 * This class is based on the Unzer Pay Later OpenAPI specification, version 1.0.0.
 */
package com.unzer.paylater.model;

public enum PaymentMethod {

    DIRECT_DEBIT("DIRECT_DEBIT"),
    BANK_TRANSFER("BANK_TRANSFER");

    private String value;

    PaymentMethod(String value) {
        this.value = value;
    }

    public static PaymentMethod fromValue(String value) {
        for (PaymentMethod b : PaymentMethod.values()) {
            if (b.value.equals(value)) {
                return b;
            }
        }
        throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }
}

